src/main/kotlin/com/demonwav/mcdev/i18n/sorting/template.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n.sorting
+
+import com.intellij.openapi.application.PathManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.project.stateStore
+import java.io.File
+
+sealed class TemplateElement
+
+data class Comment(val text: String) : TemplateElement()
+
+data class Key(val matcher: Regex) : TemplateElement()
+
+object EmptyLine : TemplateElement()
+
+data class Template(val elements: List<TemplateElement>) {
+    companion object {
+        fun parse(s: String): Template =
+            Template(
+                if (s.isNotEmpty()) {
+                    s.split('\n').map {
+                        when {
+                            it.isEmpty() -> EmptyLine
+                            it.startsWith('#') -> Comment(it.substring(1).trim())
+                            else -> Key(parseKey(it.trim()))
+                        }
+                    }
+                } else {
+                    listOf()
+                }
+            )
+
+        private val keyRegex = "([?!]?[+*]?)([^+*!?]*)([?!]?[+*]?)".toRegex()
+
+        private fun parseKey(s: String) =
+            keyRegex.findAll(s).map {
+                parseQuantifier(it.groupValues[1]) + Regex.escape(it.groupValues[2]) + parseQuantifier(it.groupValues[3])
+            }.joinToString("", "^", "$").toRegex()
+
+        private fun parseQuantifier(q: String?) =
+            when (q) {
+                "!" -> "([^.])"
+                "!+" -> "([^.]+)"
+                "!*" -> "([^.]*)"
+
+                "?" -> "(.)"
+                "?+" -> "(..+)"
+
+                "+", "?*" -> "(.+)"
+                "*" -> "(.*?)"
+
+                else -> ""
+            }
+    }
+}
+
+object TemplateManager {
+    const val FILE_NAME = "minecraft_localization_template.lang"
```

Comment:
According to IntelliJ, `FILE_NAME`, `globalFile()`, and `projectFile()` can be `private`.